dali/python/nvidia/dali/external_source.py

```diff
@@ -360,6 +360,11 @@ class ExternalSource:
     queued. If no stream is provided, DALI will use a default, with a best-effort approach at
     correctness. See the ``cuda_stream`` argument documentation for more information.
 
+    .. note::
+        After restoring from checkpoint, queries to sources that are single-argument callables
+        (accepting :class:`nvidia.dali.types.SampleInfo`, :class:`nvidia.dali.types.BatchInfo`
+        or batch index) will be resumed from the epoch and iteration saved in the checkpoint.
```

Comment:
Do we still allow to checkpoint with other sources, or should we detect them and raise an error, that they are not checkpointable, and only those single-argument callbacks are?

Reply:
I'd be more for issuing a warning: we don't want to prevent people from working around that limitation with custom initialization of the iterator, e.g. at the given epoch start.

Reply:
Warning added in https://github.com/NVIDIA/DALI/pull/5213/commits/ebcf303d5ff72637eae714f31f8071455db79dfe